### PR TITLE
prealloced plan aggregation_push_down

### PR DESCRIPTION
**Pull request by zhengwanbo**
modify constant_propagation and aggregation_push_down, prealloced space . @shenli @coocood PTAL

### CODE REVIEW

**Comment by shenli** [comment]
Related with #5069

**Comment by shenli** [comment]
@zhengwanbo Please resolve the conflicts.

File: plan/aggregation_push_down.go [hunk 1]
```diff
@@ -260,7 +260,8 @@ func (a *aggregationOptimizer) makeNewAgg(aggFuncs []aggregation.Aggregation, gb
 		GroupByItems: expression.Column2Exprs(gbyCols),
 		groupByCols:  gbyCols,
 	}.init(a.allocator, a.ctx)
-	var newAggFuncs []aggregation.Aggregation
+
+	newAggFuncs := make([]aggregation.Aggregation, 0, len(aggFuncs)+len(gbyCols))
```
Comment (XuHuaiyu) [inline plan/aggregation_push_down.go]: `len(aggFuncs)+len(gbyCols)` may be extracted as a variable which can be reused by line265.

Comment (zhengwanbo) [reply plan/aggregation_push_down.go]: great.

**Comment by zhengwanbo** [comment]
aggregation_push_down, resolved conflict successfully. PTAL again.

**Comment by XuHuaiyu** [comment]
file .DS_Store is unnecessary.

**Pull request merged**